crates/polars-core/src/chunked_array/object/extension/mod.rs

```diff
@@ -54,9 +63,9 @@ pub(crate) fn create_extension<I: Iterator<Item = Option<T>> + TrustedLen, T: Si
     iter: I,
 ) -> PolarsExtension {
     let env = "POLARS_ALLOW_EXTENSION";
-    std::env::var(env).unwrap_or_else(|_| {
-        panic!("env var: {env} must be set to allow extension types to be created",)
-    });
+    if !(std::env::var(env).is_ok() || POLARS_ALLOW_EXTENSION.load(Ordering::Relaxed)) {
```

Comment:
I think we should swap the order of these. Loading the static is likely cheaper and is what we always do in python Polars. Otherwise we must always evaluate the second branch.